pkg/storage/s3/s3.go

```diff
@@ -35,7 +35,7 @@ type Storage struct {
 func New(s3Conf *config.S3Config, timeout time.Duration, options ...func(*aws.Config)) (*Storage, error) {
 	const op errors.Op = "s3.New"
 
-	awsConfig := &aws.Config{}
+	awsConfig := defaults.Config()
```

Comment:
I'm not sure what the use case is for someone not using `UseDefaultConfig`. this PR seems to just automatically set `UseDefaultConfig` to `true` regardless of what someone sets, right?

Reply:
👋🏼 Any updates here? Not too familiar with AWS to know for sure on my side.

Reply:
@marwan-at-work me neither.

>Perhaps, consider using default always UseDefaultConfig is specified.

@manugupt1 can you explain what you mean here? Since `session.NewSession()` [reads from defaults, among other places](SDK defaults, config files, environment, and user provided config files), I'm not sure if using a `defaults.Config()` is different from just passing an empty config to `session.NewSession()`, as we were doing after this change.

Reply:
Oh sorry! All I meant was now,

defaults are not built from an empty objects. We can have some default overrides in the non-default configuration.
So, when a non-default configration `!s3Conf.UseDefaultConfiguration` is used. We keep the previous code the same.

```
	if !s3Conf.UseDefaultConfiguration {
                awsConfig := &aws.Config{}
                awsConfig.Region = aws.String(s3Conf.Region)
                for _, o := range options {
		   o(awsConfig)
	        }
	        credProviders := ...
	}

```

Reply:
@manugupt1 ok, I think I understand but still can't tell if you think something needs to change still, sorry. let me know...

Reply:
I do not think anything needs to be changed for now. Sorry for the confusion.

Reply:
ok, sounds good. I'm gonna merge this. thanks @manugupt1 !